lib/modules/side_menu/side_menu_view.dart

```diff
@@ -0,0 +1,43 @@
+part of 'side_menu_module.dart';
+
+abstract class SideMenuViewDelegate {
+  BehaviorSubject<void> get logoutButtonDidTap;
+}
+
+abstract class SideMenuView extends View<SideMenuViewDelegate> {
+  void setUser(UserInfo user);
+}
+
+class SideMenuViewImpl extends StatefulWidget {
+  const SideMenuViewImpl({Key? key}) : super(key: key);
+
+  @override
+  _SideMenuViewImplState createState() => _SideMenuViewImplState();
+}
+
+class _SideMenuViewImplState
+    extends ViewState<SideMenuViewImpl, SideMenuModule, SideMenuViewDelegate>
+    implements SideMenuView {
+  final _user = BehaviorSubject<UserInfo>();
```

Comment:
I use this to refresh UI purpose only
It means the outside only care about using `view.setUser()` and doesn't care how the UI updates

```dart
Widget build(BuildContext context) {
  return StreamSelector<UserInfo>(
    stream: _user,
   builder: (user) => Text(user.name)
  )
}
```

Reply:
I know, but shouldn't we put `_user` to presenter as that is the place to update user info then the view can listen the changes and update UI correctly?

Reply:
@manh-t I try to make my view doesn't depend on the others, it means it will provide input and output only and it can work no matter who controller it, the controller can be presenter, view controller...

Reply:
@markgravity nah, you are receiving user info from api, so the place you should get user info is from presenter, not from the view itself. If you want to create fake user info temporarily, you can create it inside presenter and emit data to the view.

Reply:
@manh-t so that `view` can't use a model object? 🤔

Reply:
@markgravity `view` can use model object but it must not hold object 😅

Reply:
@manh-t then why doest it can't hold an object? 😓 
Cause it may outupdate or memory issue?

Reply:
@markgravity if you create an object and use it internally in `view`, it's okay. But you are using the object that is received from api, so holding object inside presenter to avoid `view` changing its properties accidentally is better 🙆‍♂️

Reply:
@manh-t  ahh forgot this is reference type... 😓 
How about if I pass a copy of it?
```dart
view.setUser(user.copy())
```

Reply:
oh wait a minute, even if I do that `presenter.user`, we still send a reference of `UserInfo` to view, right?
```dart
presenter.user.add(aUser)

view.presenter.user.listen((user) {
   user.name = "aaa"
})
```

So that the root of the problem here that we should define our model with `final` properties
```dart
class UserInfo {
  final name;
  ...
}
```

Reply:
we can use `presenter.user` to get user info, no need to pass a copy imo

Reply:
@manh-t I mean, so that if I set `final` to my `UserInfo` so that I can free to use `view.setUser()` and store UserInfo in my view

Reply:
make sure that you're not violating the architecture you're implementing:
https://medium.com/cr8resume/viper-architecture-for-ios-project-with-simple-demo-example-7a07321dbd29#:~:text=VIPER%20is%20a%20backronym%20for,whatever%20the%20presenter%20tells%20it.

>View: The responsibility of the view is to send the user actions to the presenter and shows whatever the presenter tells it.

if you already had the user state from the presenter, then I don't see the reason that View stores another UserInfo for itself. You will be confused later whether which one is the source of truth.

Reply:
let's skip it, Mark will focus to solve other parts first 😄